tools/suntools/logs.py

```diff
@@ -0,0 +1,366 @@
+#!/usr/bin/env python3
+# -----------------------------------------------------------------------------
+# Programmer(s): Cody Balos and David J. Gardner @ LLNL
+# -----------------------------------------------------------------------------
+# SUNDIALS Copyright Start
+# Copyright (c) 2002-2024, Lawrence Livermore National Security
+# and Southern Methodist University.
+# All rights reserved.
+#
+# See the top-level LICENSE and NOTICE files for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+# SUNDIALS Copyright End
+# -----------------------------------------------------------------------------
+# Module of Python functions that may be useful to SUNDIALS developers writing
+# scripts to parse logs produced by SUNLogger.
+# -----------------------------------------------------------------------------
+
+import re
+import numpy as np
+from collections import ChainMap
+
+
+def convert_to_num(s):
+    """Try to convert a string to an int or float"""
+    try:
+        return np.longlong(s)
+    except ValueError:
+        try:
+            return np.double(s)
+        except ValueError:
+            return s
+
+
+def parse_logfile_payload(payload, line_number, all_lines, array_indicator="(:)"):
```

Comment:
I think the current implementation is perfectly fine, but in the future if log files become more complex, something like [pyparsing](https://github.com/pyparsing/pyparsing) would be worth considering